### ISSUE TITLE
Can't scroll using keyboard on Gatsby Dcos pages

### ISSUE THREAD

**Issue by dance2die** [gatsbyjs/gatsby#26747]
## Description

Can't scroll Gatsby docs using keyboard (arrow, home/end page up/down) keys  
while only scrolling with mouse wheel works.

This happens in both Chrome & FireFox (latest versions) Refer to `Environment` for version.

### Steps to reproduce

1. Go to either a Docs pages: https://www.gatsbyjs.com/docs/
2. Scroll the content using keyboard keys such as arrow, home/end or page up/down keys.

### Expected result

Should be able to scroll up & down

### Actual result

The docs page doesn't scroll with keyboard keys

### Environment

> Run `gatsby info --clipboard` in your project directory and paste the output here.

`gatsby` CLI is Irrelevant so I will post `npx envinfo` content instead:

```bash
$  npx envinfo 

  System:
    OS: Linux 5.4 Ubuntu 20.04.1 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz
    Memory: 1.07 GB / 15.30 GB
    Container: Yes
    Shell: 5.8 - /usr/bin/zsh
  Browsers:
    Chrome: 85.0.4183.83
    Firefox: 80.0
```

**Comment by LekoArts** [comment]
Hello, thanks for the issue!

As the main content of the page is styled with CSS Grid only the `<main>` element has the scroll (`overflow: auto`). As such not the overall page scrolls but that specific region/element. If you press <kbd>Tab</kbd> and then skip to the main content, you'll be able to scroll.

I'll ask our a11y experts if that is problematic and if yes, how we can improve it.

**Comment by dance2die** [comment]
Thank you for the review & the follow up @LekoArts 
as well as the workaround.

I will monitor this issue to see the progress :smile:

**Comment by LekoArts** [comment]
So I talked to @madalynrose and accessibility-wise it's okay as you can reach the main content via the "skip to content" link and then can scroll in it (and won't be locked into it). But I understand that from an UX standpoint the desired behavior is being able to scroll instantly. While I'll look into how we can solve this (still with a11y in mind), feel free to give suggestions on what to change -- but I don't really intend to change the CSS Grid layout in general, only as a last resort.

**Comment by machineghost** [comment]
>I'll ask our a11y experts if that is problematic and if yes, how we can improve it.

>So I talked to @madalynrose and accessibility-wise it's okay as you can reach the main content

I just want to say that I hate the "this is just an ally problem" sentiment in this thread.  Even if it was ... it's *not* acceptable to just say "screw disabled people" :(

But just as importantly, this issue *does not* just affect disabled people! It affects *anyone* trying to read the Gatsby documentation!

**Comment by alexfornuto** [comment]
FYI when I load [this doc](https://www.gatsbyjs.com/docs/adding-search-with-algolia/) (and others I assume) and hit tab, I'm not given the skip to content button, but focus is given to the current page in the ToC on the left. Scrolling with the keyboard is still not working.

**Comment by machineghost** [comment]
I found the problem.  If you look at this div, `body #___gatsby #gatsby-focus-wrapper div.css-cggu6k div.css-h0yki0`, which has the CSS:

```
    display: grid;
    position: relative;
    grid-template-rows: 1fr;
    overflow-y: auto;
    grid-template-columns: 1fr;
    max-height: calc(100vh - 0px - 84px);
```

You'll see that it also has a bunch of media queries that set alternative `max-height` values, eg.

```
@media screen and (min-width: 1000px)
.css-h0yki0 {
    grid-template-columns: 18rem 1fr;
    max-height: calc(100vh - 0px - 124px);
}
```

If you simply remove all the `max-height`s, and replace them with a simple:

    max-height: 100%

it fixes this issue, and appears to have no ill effects at any screen size (although I certainly didn't test extensively).

**Comment by gillkyle** [comment]
Thanks for the lead @dance2die! I'm afraid using `max-height` won't work in every case since we use the sidebar on many pages like /plugins, /docs, /contributing and a doc page that extends past the sidebar will leave a blank space on the left of the page.

![image](https://user-images.githubusercontent.com/21114044/97360988-8d383780-1864-11eb-91ce-3a65c8014f62.png)

If we floated the sidebar left or made it sticky we could maybe work that into a solution. The way it's using CSS grid now makes that a little harder to refactor.

The Skip to Main Content ought to be fixed up and that will be unrelated to the layout for the most part. Going to add another internal ticket to track that.

**Comment by machineghost** [comment]
First off, it was me, not @dance2die, who suggested `max-height`.

Second, on the plugin page, if you add the `max-height` as I described, and then simply add `min-height: 100%;` to the left-side nav:

    <nav aria-label="Plugin navigation" class="css-kjzo05">

It looks good to me.

**Comment by gillkyle** [comment]
Oh no, I'm sorry @machineghost I got the wrong user tagged, I apologize! I appreciate you looking into this as much as you have. I'll give that a shot now too! If it could really be that simple there'd be no reason not to do it!

**Comment by gillkyle** [comment]
It looks like we'll run into a similar problem, just a little further down the page, because the sidebar height won't fill the whole height. It's possible I'm still missing something though 🤔 I think this might need to be a bigger refactor. 

![image](https://user-images.githubusercontent.com/21114044/97367092-215acc80-186e-11eb-9c24-da832fd7a524.png)

**Comment by machineghost** [comment]
I see.  I tried playing with things for awhile, but it seemed impossible to get both: A) independently scrolling left and right blocks, **and** B) functional keyboard scrolling, without a significant overhaul.  BTW, whoever does that overhaul should really just use a flexbox, as a CSS grid seems like an appropriate tool for this (flexboxes are for laying things out in one direction, grids are for laying them out in two directions ... but you only have two "cells").

Personally I'd vote for adding the two styles I mentioned anyways, as I'd rather have a bit of white space showing below my loaded plug-ins window, and be able to keyboard scroll.

The keyboard scroll affects usability, whereas the whitespace only affects aesthetics, and only a tiny bit.  But that's just me.

**Comment by gillkyle** [comment]
Yeah that's a very valid point @machineghost. Perhaps luckily for us @fk was already working on this, and may have solved things 🎉  follow along over here: https://github.com/gatsby-inc/mansion/pull/9844#issuecomment-718767888
